### ISSUE TITLE
[RC 6.1] Bolus calculator treatments (carbs+insulin) are separated into 2 treatment entries with different timestamps

### ISSUE THREAD

**Issue by dideldum** [MilosKozak/AndroidAPS#1347]
If you enter carbs into the bolus calculator, 2 treatment entries are generated (before there was one entry with carbs and insulin togeter). While it is good to have 2 entries (the carb entry could be deleted more easily to substitute it with a corrected entry, if the carb estimation was wrong), the timestamp of the entries should not differ.
See in the attached screenshot, that the carb entry is 2 minutes before the insulin entry, though it was only one action with the bolus calculator.
Furthermore, what is very strange, that suddenly all these entries are marked as "PH", which I read means "Pump History". But they were entered in AAPS in the bolus calculator, so why do they appear, as if they were entered in the pump and read by AAPS from the pump?
In the screen shot the 2 entries at 08:08 (2.7u) and 08:06 (30g) were entered as one action in the bolus calculator and the 2 entries at 08:03(2.7u) and 08:01 (30g) were also entered as one action in the bolus calculator.
![screenshot_20180819-081016 1](https://user-images.githubusercontent.com/1267238/44308020-d13e7080-a3ad-11e8-8012-5346b7c012bc.png)

**Comment by AdrianLxM** [comment]
"PH" means that the carbs and insulin are read from the pump history.
If I remember correctly you switched from an R to an RS. The RS driver can read the pump history. This has the advantage that AAPS knows COB and IOB even after a loop break and can start right away.

Can you check how the data is presented in the pump history?

**Comment by dideldum** [comment]
Yes, it's a RS connected to AAPS. But there was no disconnect visible, the phone was close to the pump during the bolus calculator usage, the insulin delivery and after that during the meal.
I have to check the pump entries, but I can't in the next hours (daughter is not here at the moment).

**Comment by AdrianLxM** [comment]
I only do have an RS with an older firmware version (pre series model), I'm afraid.
The new firmware version should have improved history handling.

To me it would make sense (and we would have to accept it) that the pump reports the carbs as when they are entered and the bolus with the time-stamp when the delivery has finished.

This is pump-reported time. So a short disconnection during delivery with subsequent re-connection to read the history should not have any impact. At the same time it should not have much influence on the APS algorithm.

It would be nice if you could check. Thanks
